### PR TITLE
shift: Updating documentation to be more clear

### PR DESCRIPTION
**Pull request by ShiraazMoollatjie**
If you don't consult someone with knowledge of shift, it is possible to misunderstand the usage of the shift API. This PR makes it easier to reason with building FSMs

### CODE REVIEW

File: builder.go [hunk 1]
```diff
@@ -20,30 +20,30 @@ type builder FSM
 type initer builder
 
 // Insert returns a FSM builder with the provided insert status.
-func (c initer) Insert(st Status, inserter Inserter, next ...Status) builder {
+func (c initer) Insert(st Status, stInserter Inserter, nextAllowed ...Status) builder {
```
Comment (corverroos) [inline builder.go]: more verbose parameter names are not clearer. Rather improve the godoc.

File: shift.go [hunk 2]
```diff
@@ -1,10 +1,39 @@
...
+//        Update(PendingStatus, pendingReq{}, errorStatus, authorizedStatus).
+//        Update(AuthorizedStatus, authorizedReq{}, errorStatus).
+//        Update(ErrorState,  errorReq{}, PendingStatus).
+//        Build()
+//
+// It is best to avoid the following interpretation when defining a FSM:
+//
+//        FromState, UpdaterForTargetState, TargetStates
+//
+// This is an example that follows this thought pattern (it is INCORRECT!):
```
Comment (corverroos) [inline shift.go]: This is duplicated, you already gave this example above.

Also, I suggest extracting examples using proper godoc examples. See https://godoc.org/github.com/fluhus/godoc-tricks#pkg-examples

Also, suggesting making the example simpler and clearer and more generic for explanation of what it is suppoed to do; not copying latest production code you worked on ;)

File: shift.go [hunk 3]
```diff
@@ -1,10 +1,39 @@
...
 // domain model with validation, explicit fields and reflex events per state change.
 //
+// Defining an FSM
+//
 // shift.NewFSM builds a FSM instance that allows specific mutations of
 // the domain model in the underlying sql table via inserts and updates.
 // All mutations update the status of the model, mutates some fields and
 // inserts a reflex event.
+//
+// There needs to be at least one insert status defined and 0..n update
```
Comment (corverroos) [inline shift.go]: Suggestion:

A shift state machine is defined by multiple states. States have the following properties:
 - A state has a name; defined as a Status enum instance.
 - A state has an associated data struct defining the data required to enter the state.
 - A state has an allowed set of next states.
 - Only one state can be the initial state.
 - All other states are reached by explicit transitions from the initial state.
 - It is allowed to transition to a previous state or even to itself.
 - The states and transitions form a cyclical directed graph with a single root node.

**Comment by ShiraazMoollatjie** [comment]
Closing because there's an internal request to update documentation.

**Pull request closed without merging**